app/api/items/save_grade.go

```diff
@@ -175,6 +158,43 @@ func saveGradingResultsIntoDB(store *database.DataStore, user *database.User,
 	return validated, keyObtained, true
 }
 
+func saveNewScoreIntoUserAnswer(store *database.DataStore, user *database.User,
```

Comment:
🤔
Actually, the idea was to fetch the score only for the logs so if there is an error. From what I understand from the code, you do the select in every case... while the goal was to do it only if there were no row affected by the update (that may be caused by a different score or not)
So you should do the update as before, and only if there is no row affected, fetch the score.

Reply:
fixed